x-pack/plugins/monitoring/server/alerts/thread_pool_rejections_alert.ts

```diff
@@ -0,0 +1,429 @@
+  shouldFireSearchRejections?: boolean;
+  shouldFireWriteRejections?: boolean;
+}
+
+type ThreadPoolAlertData = Omit<AlertData, 'shouldFire'> & RejectionsShouldFire;
+
+interface ThreadpoolParamDetails {
+  search: CommonAlertParamDetail;
+  write: CommonAlertParamDetail;
+  [key: string]: CommonAlertParamDetail;
+}
+
+export class ThreadPoolRejectionsAlert extends BaseAlert {
+  public static readonly PARAM_DETAILS: ThreadpoolParamDetails = {
+    search: {
+      label: i18n.translate(
+        'xpack.monitoring.alerts.searchRejection.paramDetails.threshold.label',
+        {
+          defaultMessage: `SEARCH rejection`,
+        }
+      ),
+    },
+    write: {
+      label: i18n.translate('xpack.monitoring.alerts.writeRejection.paramDetails.threshold.label', {
+        defaultMessage: `WRITE rejection`,
+      }),
+    },
+  };
+  public static paramDetails = ThreadPoolRejectionsAlert.PARAM_DETAILS;
+  public static readonly TYPE = ALERT_THREAD_POOL_REJECTIONS;
+  public static readonly LABEL = i18n.translate(
+    'xpack.monitoring.alerts.threadPoolRejections.label',
+    {
+      defaultMessage: 'Thread Pool Rejections',
+    }
+  );
+
+  private readonly THREAD_POOL_TYPE = {
+    search: 'SEARCH',
+    write: 'WRITE',
+  };
+
+  public type = ThreadPoolRejectionsAlert.TYPE;
+  public label = ThreadPoolRejectionsAlert.LABEL;
+
+  protected defaultParams: ThreadPoolRejectionsAlertParams = {
+    search: {
+      threshold: 0,
+      enabled: true,
+    },
+    write: {
+      threshold: 0,
+      enabled: true,
+    },
+  };
+
+  protected actionVariables = [
+    {
+      name: 'count',
+      description: i18n.translate(
+        'xpack.monitoring.alerts.threadPoolRejections.actionVariables.count',
+        {
+          defaultMessage: 'The number of nodes reporting high thread pool rejections.',
+        }
+      ),
+    },
+    {
+      name: 'threadPoolType',
+      description: i18n.translate(
+        'xpack.monitoring.alerts.threadPoolRejections.actionVariables.threadPoolType',
+        {
+          defaultMessage: 'Rejected thread pool type eg: SEARCH, WRITE, and etc.',
+        }
+      ),
+    },
+    ...Object.values(AlertingDefaults.ALERT_TYPE.context),
+  ];
+
+  protected async fetchData(
+    params: ThreadPoolRejectionsAlertParams,
+    callCluster: any,
+    clusters: AlertCluster[],
+    uiSettings: IUiSettingsClient,
+    availableCcs: string[]
+  ): Promise<ThreadPoolAlertData[]> {
+    let esIndexPattern = appendMetricbeatIndex(this.config, INDEX_PATTERN_ELASTICSEARCH);
+    if (availableCcs) {
+      esIndexPattern = getCcsIndexPattern(esIndexPattern, availableCcs);
+    }
+
+    const { search, write } = params;
+    const { threshold: searchRejectionsThreshold, enabled: searchRejectionsEnabled } = search;
+    const { threshold: writeRejectionsThreshold, enabled: writeRejectionsEnabled } = write;
+    const checkSearchRejections = searchRejectionsEnabled && searchRejectionsThreshold > 0;
```

Comment:
It's strange that the UI will allow me to use `0` and enable this type, but the alert will not work properly on it. We should probably limit the UI so we can't use 0?